teku/src/main/java/tech/pegasys/teku/cli/deposit/EncryptedKeystoreWriter.java

```diff
@@ -45,35 +45,33 @@ public EncryptedKeystoreWriter(
     this.validatorKeyPassword = validatorKeyPassword;
     this.withdrawalKeyPassword = withdrawalKeyPassword;
     this.outputPath = outputPath;
+    createKeystoreDirectory();
   }
 
   @Override
   public void writeKeys(final BLSKeyPair validatorKey, final BLSKeyPair withdrawalKey)
       throws UncheckedIOException, KeyStoreValidationException {
-    final Path keystoreDirectory = createKeystoreDirectory(validatorKey);
 
     final KeyStoreData validatorKeyStoreData =
         generateKeystoreData(validatorKey, validatorKeyPassword);
     final KeyStoreData withdrawalKeyStoreData =
         generateKeystoreData(withdrawalKey, withdrawalKeyPassword);
 
     final String validatorFileName =
-        "validator_" + shortPublicKey(validatorKey.getPublicKey()) + ".json";
+        shortPublicKey(validatorKey.getPublicKey()) + "_validator.json";
     final String withdrawalFileName =
-        "withdrawal_" + shortPublicKey(withdrawalKey.getPublicKey()) + ".json";
+        shortPublicKey(withdrawalKey.getPublicKey()) + "_withdrawal.json";
```

Comment:
this needs to be the validator public key (short) so that we can associate the 2

Reply:
fixed